src/screens/Dashboard/Dushboard.tsx

```diff
@@ -1,88 +1,117 @@
+import AddIcon from '@mui/icons-material/Add';
 import React, { SetStateAction, useState } from 'react';
 import { Link } from 'react-router-dom';
-import AddIcon from '@mui/icons-material/Add';
 import { Layout } from '../../shared/Layout';
-import { FilterMod, ViewMod } from './Dushboard.models';
-
+import { DushboardFilterMod, DashboardDataView } from './Dushboard.models';
 import {
-  ToggleButton,
-  ToggleButtonGroup,
   Box,
   Fab,
   Typography,
   Button,
+  useTheme,
+  useMediaQuery,
 } from '@mui/material';
-
 import { DushBoardTableView } from './DushboardTableView';
+import { DushBoardCalendarView } from './DushboardCalendarView';
+import { DushboardTogglePannel } from './DushboardTogglePannel';
+import moment, { Moment } from 'moment';
+import { DushboardDateControl } from './DushboardDateControl';
 
-export default function Dashboard() {
-  const [filter, setFilter] = useState(FilterMod.ACTUAL);
-  const [selectedView, setSelectedView] = useState(ViewMod.TABLE_VIEW);
+const Dashboard = () => {
+  const [selectedDate, setSelectedDate] = useState<Moment>(moment());
+  const [filter, setFilter] = useState(DushboardFilterMod.ACTUAL);
+  const [selectedView, setSelectedView] = useState(
+    DashboardDataView.TABLE_VIEW
+  );
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
```

Comment:
I think it's good to move this piece of code in a hook as it's used in multiple places.